File: framework/core/src/python/kungfu/product_identity.py
# SPDX-License-Identifier: Apache-2.0

SECONDARY_SOURCE_SIGNATURE = "Kungfu UNGFU™"
SOURCE_PRINCIPLE = "Never Guess. Facts Unfold."


def version_banner(version: str) -> str:
    """Keep the compatible version first line and append the source signature."""
    return f"{version}\n{SECONDARY_SOURCE_SIGNATURE} · {SOURCE_PRINCIPLE}"

File: framework/gui/src/main/product-identity.ts
// SPDX-License-Identifier: Apache-2.0

export const PRODUCT_NAME = 'Kungfu';
export const SECONDARY_SOURCE_SIGNATURE = 'Kungfu UNGFU™';
export const SOURCE_PRINCIPLE = 'Never Guess. Facts Unfold.';

export function versionFirstLine(output: string): string {
  return (
    output
      .split(/\r?\n/u)
      .find((line) => line.trim().length > 0)
      ?.trim() || ''
  );
}

export function productAboutPanelOptions(applicationVersion: string) {
  return {
    applicationName: PRODUCT_NAME,
    applicationVersion,
    version: applicationVersion,
    credits: `${SECONDARY_SOURCE_SIGNATURE}\n${SOURCE_PRINCIPLE}`,
    website: 'https://kungfu.tech',
  };
}
